src/com/facebook/buck/maven/Pom.java

```diff
@@ -114,27 +137,212 @@ public Artifact apply(HasMavenCoordinates input) {
     updateModel(artifact, deps);
   }
 
-  private Model constructModel() throws IOException {
+  private Model constructModel(File file, @Nullable Model model) {
+    ModelBuilder modelBuilder = MODEL_BUILDER_FACTORY.newInstance();
+
+    try {
+      ModelBuildingRequest req = new DefaultModelBuildingRequest().setPomFile(file);
+      ModelBuildingResult modelBuildingResult = modelBuilder.build(req);
+
+      Model constructed = Preconditions.checkNotNull(modelBuildingResult.getRawModel());
+
+      return merge(model, constructed);
+    } catch (ModelBuildingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private Model constructModel() {
     File file = path.toFile();
+    Model model = new Model();
+    model.setModelVersion(POM_MODEL_VERSION);
+
+    if (publishable.getPomTemplate().isPresent()) {
+      model = constructModel(publishable.getPomTemplate().get().toFile(), model);
+    }
+
     if (file.isFile()) {
-      ModelBuildingRequest modelBuildingRequest = new DefaultModelBuildingRequest()
-          .setPomFile(file);
-      ModelBuilder modelBuilder = MODEL_BUILDER_FACTORY.newInstance();
-      try {
-        ModelBuildingResult modelBuildingResult = modelBuilder.build(modelBuildingRequest);
+      Model fromFile = constructModel(file, model);
+      model = merge(model, fromFile);
+    }
+
+    return model;
+  }
+
+  private Model merge(Model first, @Nullable Model second) {
```

Comment:
I can't find a public method that does this. Most irksome. It's not regular enough to use the JavaBeans APIs to introspect over it either.